### PR TITLE
lib: Take advantage of section descriptions

### PR DESCRIPTION
**Pull request by infinisil**
## Description of changes

See https://github.com/nix-community/nixdoc/releases/tag/v2.6.0

This work is sponsored by [Antithesis](https://antithesis.com/) :sparkles:

## Things done
- [x] Ensured that the descriptions render correctly in the manual.

### CODE REVIEW

File: lib/gvariant.nix [hunk 1]
```diff
@@ -1,16 +1,17 @@
+/*
+  A partial and basic implementation of GVariant formatted strings.
+  See [GVariant Format Strings](https://docs.gtk.org/glib/gvariant-format-strings.html) for details.
+
+  :::{.warning}
+  This API is not considered fully stable and it might therefore
+  change in backwards incompatible ways without prior notice.
+  :::
+*/
+
```
Comment (infinisil) [inline lib/gvariant.nix:10]: @linsui This now gets rendered in the Nixpkgs manual. Note that I changed the "Note" to a "Warning" here, does that sound good?

Comment (linsui) [reply lib/gvariant.nix:10]: Sounds good to me. :)

File: lib/attrsets.nix [hunk 2]
```diff
@@ -1,5 +1,5 @@
+/* Operations on attribute sets. */
 { lib }:
-# Operations on attribute sets.
```
Comment (bjornfor) [inline lib/attrsets.nix:2]: Why doesn't nixdoc support `#` as a comment character? A design choice?

Comment (infinisil) [reply lib/attrsets.nix:2]: Yeah, I think this was used to be able to distinguish between doc comments and other comments. By using `/*` for all doc comments it's clear what's gonna get rendered.

I guess this won't be a concern anymore once [RFC 145](https://github.com/NixOS/rfcs/pull/145) is implemented in nixdoc though :)

**Pull request merged**